### PR TITLE
init MetricsAdvisor

### PR DESCRIPTION
**Pull request by bowgong**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of “no breaking changes
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [ ] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  - Adding a new service

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from API Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable version with Breaking Change Validation errors
- [ ] Updating API(s) in preview over 1 year

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: specification/cognitiveservices/data-plane/MetricsAdvisor/preview/v1.0/MetricsAdvisor.json [hunk 1]
```diff
@@ -0,0 +1,6312 @@
...
+          }
+        },
+        "property": {
+          "$ref": "#/definitions/AnomalyProperty"
+        }
+      }
+    },
+    "AnomalyResultList": {
+      "required": [
+        "@nextLink",
```
Comment (johanste) [inline specification/cognitiveservices/data-plane/MetricsAdvisor/preview/v1.0/MetricsAdvisor.json:3420]: `@nextLink` should be optional (and not sent if there is no next page)

Comment (bowgong) [reply specification/cognitiveservices/data-plane/MetricsAdvisor/preview/v1.0/MetricsAdvisor.json]: fixed.

**Comment by bowgong** [comment]
@johanste , for errors in ModelValidation and SemanticValidation, I think they are all related to the polymorphism used in MA. Could you please help review and see whether it's a false alarm or needs to be fixed?

@yangyuan and @lirenhe , 
- For the Spellcheck, I've tried several methods to add custom-words.txt and/or cSpell.txt to mitigate, but none of them works. Could you please advise us?
- For the x-ms-examples, since we already have "examples" in the swagger, could you please help confirm whether x-ms-examples is still mandatory or optional?

**Comment by oscholz** [comment]
@yangyuan and @lirenhe can you please review this and help resolve the 4 failing checks with @bowgong ?

**Comment by bowgong** [comment]
Still pending on review.

**Comment by bowgong** [comment]
And it seems we have no permission to remove the no-recent-activity label.

**Comment by raych1** [comment]
@bowgong , there's a [bug](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1273422) in model validation when `hosttemplate' ends with slash. The quick work around is to remove the ending slash for [hosttemplate](https://github.com/Azure/azure-rest-api-specs/blob/eca4810a3243da34611805fc7b2cc3d16914e095/specification/cognitiveservices/data-plane/MetricsAdvisor/preview/v1.0/MetricsAdvisor.json#L5908). Or, you could suppress this error type for this swagger at readme.md.

**Comment by raych1** [comment]
@bowgong , to fix spellCheck failures it needs to add word to the file at [root folder](https://github.com/Azure/azure-rest-api-specs/blob/master/custom-words.txt)

**Comment by bowgong** [comment]
Hi @yangyuan, @lirenhe, could you please help review and sign-off this PR? Please let us know if any other prerequisites are needed.

File: specification/cognitiveservices/data-plane/MetricsAdvisor/preview/v1.0/MetricsAdvisor.json [hunk 2]
```diff
@@ -0,0 +1,5921 @@
...
+          }
+        },
+        "x-ms-examples": {
+          "Delete anomaly alerting configuration": {
+            "$ref": "./examples/deleteAnomalyAlertingConfiguration.json"
+          }
+        }
+      }
+    },
+    "/alert/anomaly/configurations": {
```
[the remark this answers is not in the document]
Comment (bowgong) [reply specification/cognitiveservices/data-plane/MetricsAdvisor/preview/v1.0/MetricsAdvisor.json:165]: The list operation should be by anomalyDetection configuration, so it is under
/enrichment/anomalyDetection/configurations/{configurationId}/alert/anomaly/configurations.

File: specification/cognitiveservices/data-plane/MetricsAdvisor/preview/v1.0/MetricsAdvisor.json [hunk 3]
```diff
@@ -0,0 +1,5921 @@
...
+          }
+        },
+        "x-ms-examples": {
+          "Delete anomaly detection configuration": {
+            "$ref": "./examples/deleteAnomalyDetectionConfiguration.json"
+          }
+        }
+      }
+    },
+    "/enrichment/anomalyDetection/configurations": {
```
[the remark this answers is not in the document]
Comment (bowgong) [reply specification/cognitiveservices/data-plane/MetricsAdvisor/preview/v1.0/MetricsAdvisor.json:538]: The list operation should be by Metric, so it is under
/metrics/{metricId}/enrichment/anomalyDetection/configurations

**Review by johanste — APPROVED** [review]

**Pull request merged**